content/docs/reference/design-docs/conflict-detection/index.en.md

```diff
@@ -0,0 +1,139 @@
+ - may take time changing state
+
+The current model only considers the following two resources:
+ - **zones**, which have one state per way to traverse it
+ - **switches**, which have one state per position
+
+{{% pageinfo color="info" %}}
+Station platforms could also be resources, which would enable modeling platform use conflicts: if two tracks share a tiny platform, it may not be allowed to use both sides of the platform at once.
+{{% /pageinfo %}}
+
+### Requirements and conflicts
+
+A **requirement** describes what resource is needed by an actor, and when.
+
+There are a few types of requirements:
+ - **shareable**: the resource is required to be in a given configuration, for a given time lapse. Other shareable requirements with the same configuration can stack up.
+ - **exclusive**: the resource is required to be in a given configuration, for a given actor, for a given time lapse. This requirement cannot be combined with others.
+
+**Conflicts** are both real-life disturbances and the incompatible resource requirements which cause them. A disturbance is something that would cause a train to be unable to run as planned.
+ - route setup delays due to unavailable resources
+ - spacing conflicts due to a train catching up with another
+
+These real-life issues can be linked to conflicts, which arise from incompatible resource requirements:
+ - an exclusive requirement overlaps another requirement
+ - a shareable requirements overlaps an incompatible requirement
+ - a resource does not have time to change state between non-overlapping, yet close, requirements
+
+For example, there are **no** conflicts when:
+ - a resource has non-overlapping exclusive requirements
+ - a resource is shared by requirements which require the same configuration
+
+## Generating requirements
+
+**For conflict detection to work, resource requirements have to strictly match what's required
+to garantee that a simulated train will not be disturbed.**
+
+### Routing requirements
+
+For a train to be routed across rail infrastructure without any trouble, some resources have to
+be available for routes to set in time. A route does not set in time if a train is disturbed by
+a route being set too late.
+
+A route might be set to late for a couple of reasons:
+ - one of the zones required may be unavailable
+ - a switch may be unavailable, or still moving
+
+Routing requirements are generated by the following algorithm:
+- a route calling timeline is computed, such as the train never slows down due to late route setting.
+  It can be done by finding when the driver would be constrained by signaling, and using that minus an allowance as the deadline.
+- for each zone in each route, simulate when it would be released, and thus not required anymore
+- use the route calling deadline as the requirement start time for each zone, and the release time as the requirement end. It is a shareable requirement.
+  Switch requirements use the same time bounds.
```

Comment:
```suggestion
- compute a route calling timeline, so that the train never slows down due to late route setting.
  It can be done by finding when the driver would be constrained by signaling, and using that minus an allowance as the deadline.
- for each zone in each route, simulate when it would be released, and thus not required anymore
- use the route calling deadline as the requirement start time for each zone, and the release time as the requirement end. It is a shareable requirement.
  Switch requirements use the same time bounds.
```
If you use "simulate" and "use", the first bullet point must follow the same conjugation